### PR TITLE
Event log delete

### PR DESCRIPTION
**Pull request by JasonIsAzn**
### Overview
This PR focused on the following:
- Add cloud function to delete logs. This delete the collection in both the event and user collection. 
- Backup deleted log data. (In the future, we need to make a schedular to automatically delete these logs?)

The PR also includes quick bug fixes (not hgh prority)
- Bug fix for attendance count
- Quick Bug  fix for random log out
- Remove SHPE Logo in home screen for Become a Member (quick solution going to reviist in low res ticket)

### Tickets/Issues
- #471

### CODE REVIEW

File: functions/src/events.ts
Comment (LucientZ) [inline functions/src/events.ts:1]: This is callable by anyone and should *not* be. Please evaluate a user's claims by evaluating `context.auth.token`

For example:
```js
const token = context.auth.token;
if (token.admin !== true && token.officer !== true && token.developer !== true && token.lead !== true && token.representative !== true) {
    throw new functions.https.HttpsError("permission-denied", `Invalid credentials`);
}
```

**Review by LucientZ — CHANGES_REQUESTED** [review]

File: functions/src/events.ts
Comment (LucientZ) [reply functions/src/events.ts:1]: Nevermind. I didn't realize that this was deleting one document. It'd be good to have some documentation as to what this actually does *in the code*

File: functions/src/events.ts [hunk 1]
```diff
@@ -248,3 +249,49 @@ export const addInstagramPoints = functions.https.onCall(async (data, context) =
     return { success: true };
 });
 
+
+export const eventLogDelete = functions.https.onCall(async (data, context) => {
+    const uid = data.uid || context.auth?.uid;
```
Comment (LucientZ) [inline functions/src/events.ts:254]: This is a MASSIVE security vulnerability. If any user is able to call this function, they can delete any event log they want. I've just realized that `eventSignIn` and `eventSignOut` have this vulnerability as well and will be refactoring those as soon as possible. Do *not* deploy this cloud function

**Review by LucientZ — CHANGES_REQUESTED** [review]

File: functions/src/events.ts
Comment (JasonIsAzn) [reply functions/src/events.ts:1]: Add comment to function

File: functions/src/events.ts (hunk 1 above)
Comment (JasonIsAzn) [reply functions/src/events.ts:258]: okay. undeployed the function.

**Review by LucientZ — APPROVED** [review]

**Pull request merged**